runtime/v2/config.go

```diff
@@ -0,0 +1,59 @@
+package runtime
+
+import (
+	"strings"
+
+	"cosmossdk.io/core/server"
+	"cosmossdk.io/depinject"
+)
+
+// GlobalConfig is a recursive configuration map containing configuration
+// key-value pairs parsed from the configuration file, flags, or other
+// input sources.
+type GlobalConfig server.ConfigMap
+
+// ModuleConfigMaps is a map module scoped ConfigMaps
+type ModuleConfigMaps map[string]server.ConfigMap
+
+// ProvideModuleConfigMaps returns a map of module name to module config map.
+// The module config map is a map of flag to value.
+func ProvideModuleConfigMaps(
+	moduleConfigs []server.ModuleConfigMap,
+	globalConfig GlobalConfig,
+) ModuleConfigMaps {
+	moduleConfigMaps := make(ModuleConfigMaps)
+	for _, moduleConfig := range moduleConfigs {
+		cfg := moduleConfig.Config
+		name := moduleConfig.Module
+		moduleConfigMaps[name] = make(server.ConfigMap)
+		for flag, df := range cfg {
+			m := globalConfig
+			fetchFlag := flag
+			for _, part := range strings.Split(flag, ".") {
```

Comment:
If this is a way to remove the server name part of the flag, could we just add a comment?
I had to pause to get it.

Reply:
Given that we support nested config, and both CLI and Viper syntax specify nesting with a ".", I think it needs to be supported?

Reply:
yeah, im just asking if we can add a comment here, as it isn't obvious at the first glance